R/TealAppDriver.R

```diff
@@ -575,6 +578,31 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
         export = export,
         ...
       )
+    },
+    #' @description
+    #' Check if the page is stable without any DOM updates for the specified stability period.
+    #' @param element_selector (`character(1)`) `CSS` selector to check the stability of the page.
+    #' @param stability_period (`numeric(1)`) The time in milliseconds to wait for the page to be stable.
+    #' @param check_interval (`numeric(1)`) The time in milliseconds to check for changes in the page.
+    #' The stability check is reset when a change is detected in the page after sleeping for check_interval.
+    #' Parameter with additional value to passed in `wait_for_value`.
+    wait_for_element_stability = function(element_selector = "body", stability_period = 500, check_interval = 50) {
+      element_content <- self$get_html(element_selector)
+      times_run <- 0
+      max_times_run <- stability_period / check_interval
+
+      while (TRUE) {
+        current_element <- self$get_html(element_selector)
+        if (!identical(current_element, element_content)) {
+          element_content <- current_inner_html
+          times_run <- 0
+        } else if (times_run < max_times_run) {
+          times_run <- times_run + 1
+        } else {
+          break
+        }
+        Sys.sleep(check_interval / 1000)
+      }
```

Comment:
Should we move this method to private 🤔 ?

```suggestion
      element_content <- self$get_html(element_selector)
      times_run <- 0
      max_times_run <- stability_period / check_interval

      while (TRUE) {
        current_element <- self$get_html(element_selector)
        if (!identical(current_element, element_content)) {
          element_content <- current_inner_html
          times_run <- 0
        } else if (times_run < max_times_run) {
          times_run <- times_run + 1
        } else {
          break
        }
        Sys.sleep(check_interval / 1000)
      }
```
How about using "repeat with break"? Here is an alternative time comparison logic. Feel free to ignore.
```suggestion
      element_content <- self$get_html(element_selector)
  end_time <- Sys.time() + (stability_period / 1000)
  
  repeat {
    Sys.sleep(check_interval / 1000)
    current_element <- self$get_html(element_selector)
    
    if (!identical(current_element, element_content)) {
      element_content <- current_element
      end_time <- Sys.time() + (stability_period / 1000)
    } else if (Sys.time() >= end_time) {
      break
    }
  }
```

Reply:
Sure, this one is easy to interpret as well. I did move it to private and renamed it to just handle `body` change.